vscode/src/circuitEditor.ts

```diff
@@ -0,0 +1,169 @@
+        circuit,
+      };
+
+      const message = {
+        command: "circuit",
+        props,
+      };
+      webviewPanel.webview.postMessage(message);
+    };
+
+    // Update the webview when the text document changes
+    // const changeDocumentSubscription = vscode.workspace.onDidChangeTextDocument(
+    //   (event) => {
+    //     if (event.document.uri.toString() === document.uri.toString()) {
+    //       if (
+    //         this.savedVersion !== undefined &&
+    //         event.document.version > this.savedVersion
+    //       ) {
+    //         // Update the webview with the new document content
+    //         updateWebview();
+    //       }
+    //     }
+    //   },
+    // );
+
+    // // Dispose of the event listener when the webview is closed
+    // webviewPanel.onDidDispose(() => {
+    //   changeDocumentSubscription.dispose();
+    // });
+
+    updateWebview();
+  }
+
+  private getHtmlForWebview(webview: vscode.Webview): string {
+    const extensionUri = this.context.extensionUri;
+
+    function getUri(pathList: string[]) {
+      return webview.asWebviewUri(
+        vscode.Uri.joinPath(extensionUri, ...pathList),
+      );
+    }
+
+    const katexCss = getUri(["out", "katex", "katex.min.css"]);
+    const githubCss = getUri(["out", "katex", "github-markdown-dark.css"]);
+    const webviewCss = getUri(["out", "webview", "webview.css"]);
+    const scriptUri = getUri(["out", "webview", "editor.js"]);
+    const resourcesUri = getUri(["resources"]);
+    return `
+      <!DOCTYPE html>
+      <html lang="en">
+        <head>
+          <meta charset="UTF-8">
+          <meta name="viewport" content="width=device-width, initial-scale=1.0">
+          <title>Q#</title>
+          <link rel="stylesheet" href="${githubCss}" />
+          <link rel="stylesheet" href="${katexCss}" />
+          <link rel="stylesheet" href="${webviewCss}" />
+          <script src="${scriptUri}"></script>
+          <script>
+            window.resourcesUri = "${resourcesUri.toString()}";
+          </script>
+        </head>
+        <body>
+        </body>
+      </html>`;
+  }
+
+  private getDocumentAsJson(document: vscode.TextDocument): any {
+    const text = document.getText();
+    if (text.trim().length === 0) {
+      return {};
+    }
+
+    try {
+      return JSON.parse(text);
+    } catch {
+      throw new Error(
+        "Could not get document as json. Content is not valid json",
+      );
+    }
+  }
+
+  private async updateTextDocument(
+    document: vscode.TextDocument,
+    circuit: string,
+  ) {
+    // Short-circuit if there are no changes to be made.
+    if (
+      circuit.trim().replace(/\r\n/g, "\n") ===
+      document.getText().trim().replace(/\r\n/g, "\n")
+    ) {
+      return;
+    }
+
+    const edit = new vscode.WorkspaceEdit();
+
+    // Just replace the entire document every time for this example extension.
+    // A more complete extension should compute minimal edits instead.
```

Comment:
"Example extension"? Is this left over from sample code?

Reply:
removed